ADD: rotation of an image with 90, 180, -90 degree

Winbay opened the pull request:
The simple rotate function only for 90, 180 and -90 degrees.

Winbay committed:
ADD: rotation of an image with 90, 180, -90 degree

**ADD**:
-`src/utils/bool.h`: create typedef bool
-`src/../rotate.c`: file to call when you have to rotate an image with
90, 180 or -90 degrees, it only rotate the image and apply it

**FIX**:
`Makefile`: new file and the CFLAG "-lm"

JulesdeCube commented on src/load_Image/rotate.h:
```diff
@@ -0,0 +1,32 @@
+ ** \param filter function to apply on each pixel
+*/
+void apply_rotation(GtkWidget* image, bool has_inverse_HW,
+                    guchar* filter(int, int, guchar *, int, int , int , int));
```
defimnit a new type for the function and document function parameters

JulesdeCube commented on src/load_Image/rotate.h:
```diff
@@ -0,0 +1,32 @@
+
+
+/**
+ ** \brief Switch from param degree and call the corresponding function 
```
rotate the given image by `degee` degree.

JulesdeCube commented on src/main.c:
```diff
@@ -12,6 +14,8 @@ typedef struct {
 
 int main(int argc, char *argv[])
 {
+    create_window(argc, argv);
+    /*
```
no test in the version branch

JulesdeCube commented on src/load_Image/rotate.h:
```diff
@@ -0,0 +1,32 @@
+#include <gdk/gdk.h>
+
+#include "tools.h"
+#include "../src/utils/bool.h"
```
wong path, repace by 
```suggestion
#include "../utils/bool.h"
```

JulesdeCube commented on src/load_Image/rotate.c:
```diff
@@ -0,0 +1,139 @@
+
+    //New pixbuf for an image
+    newpixbuf = gdk_pixbuf_new (GDK_COLORSPACE_RGB, 0, 8, newwidth, newheight);
+    int newRowstride = gdk_pixbuf_get_rowstride (newpixbuf);
```
??????????????????????????

JulesdeCube reviewed (commented):
need variable rotation

Winbay committed:
ADD: rotation of an image with all other degrees.
Now there is one classic rotation and 3 others type of rotation which
are more optimize

**FIX**:
-`src/../rotate.c`: add all specific functions for classical rotation

**ADD**:
-`src/../rotate.h`: new enum with 3 types of rotation.
New struct "rotate_tools" where there is all tools to make rotations.
-`src/../tools.c`: 3 new functions for maths
and one more to set 3 rgb values of a pixel (`set_3values`)

JulesdeCube commented on src/load_Image/load.c:
```diff
@@ -29,7 +29,7 @@ void activate_window (GtkApplication *app, gpointer user_data)
           HERE TO CHANGE THE TRANSFORMATION
       ################################################
     */
-    rotate(imagec, -90);
```
pas de test dans le main

JulesdeCube commented on src/load_Image/rotate.h:
```diff
@@ -7,26 +7,70 @@
+ */
+enum rotate_type
+{
+    INVERSE_HW,
```
comment here the signification of the value

JulesdeCube reviewed (changes requested):
2 error in the comments

Winbay committed:
FIX: comments and main.c

**FIX**:
-`src/../rotate.h`: change the name of the enum and the struct
-`src/main.c`: include rotate.h and remove tests calls
